tobiasHe/assignments/mergeIntervals/lc57/lc57.js

```diff
@@ -0,0 +1,111 @@
+// Problem Statement #
+
+// Given a list of non-overlapping intervals sorted by their start time, insert a given interval at the correct position and merge all necessary intervals to produce a list that has only mutually exclusive intervals.
+
+class Interval {
+	constructor(start, end) {
+		this.start = start;
+		this.end = end;
+	}
+
+	print_interval() {
+		process.stdout.write(`[${this.start}, ${this.end}]`);
+	}
+}
+
+const insert = function(intervals, new_interval) {
+	// TODO: Write your code here
+	// determine where the new interval will overlap
+	// since all the given intervals are not overlapping and sorted we can safely assume that if the start of new interval is greater than the end times
+	// then we can push the new interval after the ones that satisfy (new_interval.start > interval.end[...])
+	// find relevant intervals that the new interal affects it and insert it
+
+	let merged = [];
+	let i = 0;
+	let length = intervals.length;
+	//edge cases  if empty list of intervals just return newIntervals in the array
+	if (length < 1) {
+		merged.push(new_interval);
+		return merged;
+	}
+	//find the first place to insert our new_interval which is right before the interval it is overlapping with
```

Comment:
In-line comments make code look cluttered, so try to avoid them at all possible costs. Your approach and psuedocode can be written above the function declaration and be referenced there.